File: backend/QuizHubApplication/DTOs/Responses/ResultDetailsDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuizHubApplication.DTOs.Responses
{
    public record ResultDetailsDto(
     int Id,
     int QuizAttemptId,
     string QuizTitle,
     int TotalQuestions,
     int CorrectAnswers,
     int Score,
     double Percentage,
     int TimeTakenMin,
     string CreatedAt,
     List<ResultDetailsQuizAttemptAnswerDto> AttemptAnswerDtos



        )
    {
    }
}

File: frontend/QuizHubFrontend/src/app/features/result/models/ResultDetailsDto.ts
import { ProgressDto } from './ProgressDto';
import { ResultDetailsQuizAttemptAnswerDto } from './ResultDetailsQuizAttemptAnswerDto';

export interface ResultDetailsDto {
    id: number;
    quizAttemptId: number;
    quizTitle: string;
    totalQuestions: number;
    correctAnswers: number;
    score: number;
    percentage: number;
    maximumScore: number;
    timeTakenMin: number;
    createdAt: string;
    unanswerdQuestion: number;
    attemptAnswerDtos: ResultDetailsQuizAttemptAnswerDto[];
    progress: ProgressDto[];
}
